### PR TITLE
Changed data fetching file name to overview to fix meta data title

### LINKED ISSUE #33522: Docs: Web SDK section broken
### What is the improvement or update you wish to see?

Hello,

This will be a bit of an unsual request as it's rather a bug on documentation website.
It looks like https://nextjs.org/docs/basic-features/data-fetching#incremental-static-regeneration and https://nextjs.org/docs/basic-features/data-fetching are not working anymore.

Most probably a routing issue with the new version being released.

Hope I posted the issue in the right section :confused: 

### Is there any context that might help us understand?

Accessing https://nextjs.org/ and scrolling to `The Web SDK` of the page, there are 2 blocks that are pointing to a non existing page.
- Hybrid: SSG and SSR
- Incremental static generation

### Does the docs page already exist? Please link to it.

_No response_

### PR DESCRIPTION
**Pull request by molebox**
This PR aims to fix the meta data title on og images by changing the file path name from index to overview. It also fixes all links to this path and adds a redirect on the route

fixes #33522

## Bug

- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Errors have helpful link attached, see `contributing.md`

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have helpful link attached, see `contributing.md`

## Documentation / Examples

- [ ] Make sure the linting passes by running `yarn lint`

### CODE REVIEW

**Review by leerob — DISMISSED** [review]

File: docs/api-reference/data-fetching/get-server-side-props.md [hunk 1]
```diff
@@ -144,7 +144,7 @@ export default Page
 For more information on what to do next, we recommend the following sections:
 
 <div class="card">
-  <a href="/docs/basic-features/data-fetching/index.md">
+  <a href="/docs/basic-features/data-fetching/overviewmd">
```
Comment (styfle) [inline docs/api-reference/data-fetching/get-server-side-props.md:147]:
```suggestion
  <a href="/docs/basic-features/data-fetching/overview.md">
```

**Comment by molebox** [comment]
@styfle would you mind taking another look here please, and merging if all 🟢

**Review by ijjk — DISMISSED** [review]

**Comment by ijjk** [comment]
It looks like the manifest needs to be updated still https://github.com/vercel/next.js/runs/4896213283?check_suite_focus=true

**Comment by molebox** [comment]
> It looks like the manifest needs to be updated still https://github.com/vercel/next.js/runs/4896213283?check_suite_focus=true

Thanks @ijjk I confused how this works. This is the snippet:

```
            {
              "title": "Data Fetching",
              "path": "/docs/basic-features/data-fetching/index.md",
              "redirect": {
                "destination": "/docs/basic-features/data-fetching/overview.md",
                "permanent": true
              },
```

Do i change the original path to point to `overview.md` too?

**Review by ijjk — APPROVED** [review]

**Pull request merged**